tests/functional/states/test_kubernetes.py

```diff
@@ -420,6 +416,65 @@ def test_deployment_present_replace(kubernetes, deployment, kubernetes_exe):
     assert deployment_state["spec"]["replicas"] == 3
 
 
+def test_deployment_present_patch(kubernetes, deployment, kubernetes_exe):
```

Comment:
suggestion: I'd test in test mode as well and assert that the resource is not changed. It's often neglected when writing a state, but can have surprising consequences if not handled correctly.